tests/TestCase/ORM/Rule/LinkConstraintTest.php

```diff
@@ -190,10 +192,18 @@ public function testInvalidRepository($options): void
         $this->expectException(InvalidArgumentException::class);
         $this->expectExceptionMessage('Argument 2 is expected to have a `repository` key that holds an instance of `\Cake\ORM\Table`');
 
-        $Articles = $this->getMockForModel('Articles', ['buildRules'], ['table' => 'articles']);
-
         $rulesChecker = new RulesChecker($options);
-        $Articles->expects($this->atLeastOnce())->method('buildRules')->willReturn($rulesChecker);
+
+        $Articles = Mockery::mock(
+            ArticlesTable::class . '[buildRules]',
```

Comment:
is the `[buildRules]` necessary here?

Reply:
Yes, since that's the method we want to mock

Reply:
I thought thats what the `->shouldReceive('buildRules')` is already doing automatically

Reply:
at least in all the mockery conversion I did in the past i never added this here since it seems to me like duplicate information.

Reply:
Without specifying the method name all the class methods would get mocked if `->makePartial()` is not called.